### ISSUE TITLE
`RuntimeException` when running the JAR file

### ISSUE THREAD

**Issue by rish-16** [nus-cs2103-AY2122S1/forum#274]
Hey everyone,

I'm using IntelliJ on macOS (non M1). I have the standard working version of Java and macOS for this module. No issues there.

> My JAR file is `11.4MB` and I'm sure it contains the JavaFX material inside. Otherwise, it was originally `10KB`. 

I created a JAR file using the `Launcher` object as the main entry point. As such, when I copied the JAR file to a separate folder and run it using `java -jar ip.jar`, I get this exception in the console:

```java
Exception in Application start method
Exception in thread "main" java.lang.RuntimeException: Exception in Application start method
        at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:900)
        at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:195)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalStateException: Location is not set.
        at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2459)
        at javafx.fxml.FXMLLoader.load(FXMLLoader.java:2435)
        at duke.Main.start(Main.java:25)
        at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(LauncherImpl.java:846)
        at com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:455)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
rish $ 
```

It properly runs through the process of creating the text file (I can see it being created in my file explorer) but immediately conks out after with the above exception. Have no idea where to look or how to assess the problem.

Hoping for any help before the final deadline tonight 🤕 

Thanks!

**Comment by rish-16** [comment]
It's a pretty weird situation to be in because nowhere does it state what's wrong with my code. It's giving me issues in the libraries' code (I can assure you I haven't written 2435 LOC 😨).

Does anyone know how to debug or fix this? Not really sure what to search on Google/Stackoverflow either because the message is so cryptic.

**Comment by damithc** [comment]
> Not really sure what to search on Google/Stackoverflow either because the message is so cryptic.

@rish-16 Try `java.lang.IllegalStateException: Location is not set`

**Comment by vimuthm** [comment]
Hi,
 - I'm able to build and run your current latest version (1h ago on master) without encountering this issue, so try debugging later additions.
 - One other thing you can try (just a wild guess tbh), in case debugging the `IllegalStateException` isn't fruitful: I noticed you're running `ip.jar`, try `duke.jar` instead which should have been created along side; if I'm right this jar doesn't require specific modification of the `manifest` file as you seem to have done.

**Comment by rish-16** [comment]
Hey @VimuthM, thanks for reaching out!

May I know what it looks like if you type in `list` in the chatbot for the version you tried out? I just added a patch that lets the bot create the `duke.txt` file in case it doesn't exist. 

Also, I tried running `duke.jar` that I found in the `build/libs/duke.jar` directory. But I got these errors:

```
Error: Main method not found in class duke.Duke, please define the main method as:
   public static void main(String[] args)
or a JavaFX application class must extend javafx.application.Application
```

Not sure how to proceed from here either 🤷🏻‍♂️  I'll try something out.

Thanks, Vimuth 🙏🏻
